src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs

```diff
@@ -210,23 +213,26 @@ public int ResolveName(string? pszName, uint dwFlags, out IVsEnumDebugName? ppNa
             });
 
             return ppNames != null ? VSConstants.S_OK : VSConstants.E_NOTIMPL;
-        }
 
-        private async ValueTask<IVsDebugName> CreateDebugNameAsync(
-            BreakpointResolutionResult breakpoint, CancellationToken cancellationToken)
-        {
-            var document = breakpoint.Document;
-            var filePath = _languageService.Workspace.GetFilePath(document.Id);
-            var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
-            var span = text.GetVsTextSpanForSpan(breakpoint.TextSpan);
-            // If we're inside an Venus code nugget, we need to map the span to the surface buffer.
-            // Otherwise, we'll just use the original span.
-            var mappedSpan = await span.MapSpanFromSecondaryBufferToPrimaryBufferAsync(
-                this.ThreadingContext, document.Id, cancellationToken).ConfigureAwait(true);
-            if (mappedSpan != null)
-                span = mappedSpan.Value;
-
-            return new VsDebugName(breakpoint.LocationNameOpt, filePath!, span);
+            async ValueTask<IVsDebugName> CreateDebugNameAsync(
+                BreakpointResolutionResult breakpoint, CancellationToken cancellationToken)
+            {
+                // We're in a blocking jtf run.  So CA(true) all calls to ensure we're coming bac
+                // and using the blocked thread whenever possible.
+
+                var document = breakpoint.Document;
+                var filePath = _languageService.Workspace.GetFilePath(document.Id);
+                var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(true);
+                var span = text.GetVsTextSpanForSpan(breakpoint.TextSpan);
+                // If we're inside an Venus code nugget, we need to map the span to the surface buffer.
+                // Otherwise, we'll just use the original span.
+                var mappedSpan = await span.MapSpanFromSecondaryBufferToPrimaryBufferAsync(
```

Comment:
>MapSpanFromSecondaryBufferToPrimaryBufferAsync [](http://example.com/codeflow?start=44&length=46)

It looks like the only reason MapSpanFromSecondaryBufferToPrimaryBufferAsync is async is to ensure we're on the main thread. Should we just use a sync version of that method instead?